### PR TITLE
Fixed DECLARE_DYNAMIC_CLUSTER to support CLIENT role as well.

### LINKED ISSUE #30872: [BUG] Client role is not supported for dynamic clusters
### Reproduction steps

The `DECLARE_DYNAMIC_CLUSTER` macro has `ZAP_CLUSTER_MASK(SERVER)` role hardcoded, which makes every dynamic cluster created to work as a server. In case someone would like to create a client role and e.g. send commands to control other server devices it's not currently possible.

I tested it with binding configured however it fails on `emberAfContainsClient`.

### Bug prevalence

Whenever I do this

### GitHub hash of the SDK that was being used

4db8c38845ed915657c3dc7ff87dcf09b7efcf6d

### Platform

core

### Platform Version(s)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
The DECLARE_DYNAMIC_CLUSTER has SERVER role hardcoded, which means it is not possible to declare dynamic endpoint using a CLIENT role.

The fix allows to pass role, as an argument and use both ZAP_CLUSTER_MASK(SERVER) and ZAP_CLUSTER_MASK(CLIENT) on cluster creation.

Fixes: #30872

### CODE REVIEW

**Review by markaj-nordic — APPROVED** [review]
LGTM. Any reason why it was hardcoded?

**Comment by kkasperczyk-no** [comment]
> LGTM. Any reason why it was hardcoded?

Unfortunately, I have no clue.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
There are no client attributes in Matter...  so I am not sure what this is trying to do.

**Review by bzbarsky-apple — COMMENTED** [review]
Or is it just the PR title (and presumably commit message) that's wrong?

**Comment by kkasperczyk-no** [comment]
> Or is it just the PR title (and presumably commit message) that's wrong?

Yes, of course you are right, I meant client role for a cluster, not attribute. The commit message and PR title were fixed.

File: examples/bridge-app/esp32/README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ defined:
     application's `main.cpp` for an example of this implementation.
 
 `DECLARE_DYNAMIC_CLUSTER_LIST_BEGIN(clusterListName)`
-`DECLARE_DYNAMIC_CLUSTER(clusterId, clusterAttrs, incomingCommands, outgoingCommands)`
+`DECLARE_DYNAMIC_CLUSTER(clusterId, clusterAttrs, role, incomingCommands, outgoingCommands)`
```
Comment (bzbarsky-apple) [inline examples/bridge-app/esp32/README.md:65]: Might be worth documenting what exactly "role" is expected to be.

File: src/app/util/attribute-storage.h [hunk 2]
```diff
@@ -57,9 +57,9 @@
 
 #define DECLARE_DYNAMIC_CLUSTER_LIST_BEGIN(clusterListName) EmberAfCluster clusterListName[] = {
 
-#define DECLARE_DYNAMIC_CLUSTER(clusterId, clusterAttrs, incomingCommands, outgoingCommands)                                       \
+#define DECLARE_DYNAMIC_CLUSTER(clusterId, clusterAttrs, role, incomingCommands, outgoingCommands)                                 \
```
Comment (bzbarsky-apple) [inline src/app/util/attribute-storage.h:60]: This _definitely_ needs to clearly document what `role` is expected to be.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/util/attribute-storage.h (hunk 2 above)
Comment (kkasperczyk-no) [reply src/app/util/attribute-storage.h:62]: Ok, done.

**Review by LuDuda — APPROVED** [review]

**Pull request merged**